hera_pspec/uvpspec.py

```diff
@@ -554,91 +555,205 @@ def key_to_indices(self, key, *args):
+        # Use _select() to pick out only the requested baselines/spws
+        if just_meta:
+            _select(self, spws=spws, bls=bls,
+                    only_pairs_in_bls=only_pairs_in_bls, 
+                    blpairs=blpairs, times=times, pols=pols)
+        else:
+            _select(self, spws=spws, bls=bls, 
+                    only_pairs_in_bls=only_pairs_in_bls, 
+                    blpairs=blpairs, times=times, pols=pols, 
+                    h5file=grp)
+        
+        self.check(just_meta=just_meta)
+    
+    
+    def read_hdf5(self, filepath, just_meta=False, spws=None, bls=None,
+                  blpairs=None, times=None, pols=None, 
+                  only_pairs_in_bls=False):
         """
         Clear current UVPSpec object and load in data from an HDF5 file.
 
         Parameters
         ----------
         filepath : str, path to HDF5 file
 
-        just_meta : boolean, read-in only metadata and no data, wgts or integration arrays
-
-        spws : list of spectral window integers to select
+        just_meta : bool, optional
+            If True, read-in metadata but ignore data, wgts and integration 
+            arrays. Default: False.
 
-        bls : list of i6 baseline integers or baseline tuples, Ex. (2, 3) 
-            Select all baseline-pairs whose first _or_ second baseline are in bls list.
-            This changes if only_pairs_in_bls == True.
+        spws : list of tuple, optional
+            List of spectral window integers to select. Default: None (loads 
+            all channels).
 
-        only_pairs_in_bls : bool, if True, keep only baseline-pairs whose first _and_ second baseline
-            are found in bls list.
+        bls : list of i6 baseline integers or baseline tuples
+            Select all baseline-pairs whose first _or_ second baseline are in 
+            the list. This changes if only_pairs_in_bls == True.
+            Example tuple: (2, 3). Default: None (loads all bl pairs).
 
-        blpairs : list of baseline-pair tuples or integers to keep, if bls is also fed, this list is concatenated
-            onto the baseline-pair list constructed from from the bls selection
+        blpairs : list of baseline-pair tuples or integers
+            List of baseline pairs to keep. If bls is also fed, this list is 
+            concatenated onto the baseline-pair list constructed from from the 
+            bls selection.
  
-        times : float ndarray of times from the time_avg_array to keep
+        times : float ndarray
+            Times from the time_avg_array to keep.
+        
+        pols : list of str or int
+            List of polarization strings or integers to keep. See 
+            pyuvdata.utils.polstr2num for acceptable options.
+        
+        only_pairs_in_bls : bool, optional
+            If True, keep only baseline-pairs whose first _and_ second baseline
+            are found in the 'bls' list. Default: False.
         """
         # clear object
```

Comment:
you can erase this line b/c the `_clear()` method is called in `read_from_group`